### PR TITLE
fix(profiling): Fall back to default platform on profiling panel

### PR DESCRIPTION
**Pull request by Zylphrex**
The project's platform can be null and would crash PlatformIcon.

### CODE REVIEW

File: static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx [hunk 1]
```diff
@@ -147,7 +147,7 @@ function SlowestTransactionPanelItem({
   return (
     <PanelItem key={transaction.transaction}>
       <Flex justify="space-between" gap={space(1)}>
-        <PlatformIcon platform={transactionProject?.platform} />
+        <PlatformIcon platform={transactionProject?.platform || 'default'} />
```
Comment (evanpurkhiser) [inline static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx:150]:
```suggestion
        <PlatformIcon platform={transactionProject?.platform ?? 'default'} />
```

Is this not correctly reflected in the `PlatformIcon` types?

Comment (Zylphrex) [reply static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx:150]: Hm... It looks like types aren't being checked for `PlatformIcon`, any ideas why that is? I can pass anything as `platform` and nothing complains

Comment (evanpurkhiser) [reply static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx:150]: hmm, it should be type string only https://github.com/getsentry/platformicons/blob/master/src/platformIcon.tsx#L152

Comment (Zylphrex) [reply static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx:150]: yeah i see that, but locally, it's not complain no matter what i pass it. null/undefined/objects/number all pass the type checking

**Review by JonasBa — APPROVED** [review]
I would much rather we do this change in PlatformIcon so we have a fallback by default - I suspect this could also be an issue on other places in Sentry?

**Comment by Zylphrex** [comment]
> I would much rather we do this change in PlatformIcon so we have a fallback by default - I suspect this could also be an issue on other places in Sentry?

Agreed, will audit the rest of the code base once I update platformicons

**Pull request merged**